src/assignments/orbit-report.rst

```diff
@@ -0,0 +1,414 @@
+   * You should NOT see the message "orbit-list worked!". Why?
+   
+3. Open developer tools in your browser and look at the JavaScript console.
+
+You should see the below error message telling you that the ``orbit-list`` compoment does NOT have a ``satellites`` property.
+Note only the relavent message text has been included below.
+
+::
+
+  Error: Template parse errors:
+  Can't bind to 'satellites' since it isn't a known property of 'app-orbit-list'.
+  1. If 'app-orbit-list' is an Angular component and it has 'satellites' input, then verify that it is part of this module.
+
+To solve this issue, you need to declare in ``orbit-list.component.ts`` that the ``orbit-list`` compoment has an input property named ``satellites``.
+
+4. Add the code below to line 10 of ``orbit-list.component.ts``
+   
+   * ``@Input() satellites: Satellite[];``
+
+The ``@Input()`` is special Angular syntax that declares that ``satellites`` is a property that will be passed into the component via ``<app-orbit-list [satellites]="sourceList"></app-orbit-list>``.
+
+5. Update the ``require`` statements to import ``Input`` and ``Satellite``
+
+   * ``import { Component, OnInit, Input } from '@angular/core';``
+   * ``import { Satellite } from '../satellite';``
+
+6. View the app in your browser. You should see: orbit-list works!
+
+   * You still don't have satellites showing yet. That is the next step.
+
+Your ``orbit-list.component.ts`` should now look like the below.
+
+.. sourcecode:: typescript
+   :linenos:
+
+   import { Component, OnInit, Input } from '@angular/core';
+   import { Satellite } from '../satellite';
+
+   @Component({
+      selector: 'app-orbit-list',
+      templateUrl: './orbit-list.component.html',
+      styleUrls: ['./orbit-list.component.css']
+   })
+   export class OrbitListComponent implements OnInit {
+
+      @Input() satellites: Satellite[];
+
+      constructor() { }
+
+      ngOnInit() {
+      }
+
+   }
+
+
+4) Display Table of Satellites
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Now that ``orbit-list`` has a ``satellites`` property that is an array of ``Satellite`` objects, you can
+use that array to build an HTML table with each row being a different satellite.
+
+1. In ``orbit-list.component.html`` use``*ngFor`` to loop over the ``satellites``. The HTML table you build should look like the following.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <h3>Orbit Report</h3>
+   <table>
+      <tr class="header-row">
+         <th class="sortable">Name</th>
+         <th class="sortable">Type</th>
+         <th>Operational</th>
+         <th>Orbit Type</th>
+         <th>Launch Date</th>
+      </tr>
+      <!-- TODO: put <tr *ngFor=""></tr> here -->
+   </table>
+
+Next you need to include CSS that will make your table and application look a little nicer.
+
+2. Copy the entire contents of the `example orbit-list-component.css  <https://gist.github.com/welzie/5247f5ac36e973903cd5202af50932e6>`_ and put it into your ``orbit-list-component.css``.
```

Comment:
My vote here would be to copy the ``sortable`` class from the source files and then provide a list of requirements (e.g. center the text in each table cell, ``td``). We could also have the students research options for "making your table look nicer" instead of providing ready-made CSS. Maybe something like, "Goggle 'css alternate row color' and style your table with *zebra stripes*."

Reply:
I can't decide on how much CSS to include. How about for now I add these CSS tasks to the bonus.

Reply:
Also I removed all the `em` syntax.

Reply:
OK. You convinced me.